Automattic/jetpack
# [RNMobile] Initial Port of Jetpack Tiled Gallery Block

Fixes #[4268](https://github.com/wordpress-mobile/gutenberg-mobile/issues/4268)

#### Related PRs

* Gutenberg Mobile: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4291
* Android: https://github.com/wordpress-mobile/WordPress-Android/pull/15619
* iOS: https://github.com/wordpress-mobile/WordPress-iOS/pull/17618
* Gutenberg: https://github.com/WordPress/gutenberg/pull/37776

#### Changes proposed in this Pull Request:
* Port the Tiled Gallery to the native mobile apps using inner Image blocks. This deviates from the current web implementation – which continues to use a custom Image component – but aligns closely with the recently refactored Gallery block.
* This PR adds `.native.js` equivalents of a number of files related to the `tiled-gallery` block in order to make the block usable in the [gutenberg-mobile](https://github.com/wordpress-mobile/gutenberg-mobile) editor used by the [WordPress Android](https://github.com/wordpress-mobile/WordPress-Android) and [iOS apps](https://github.com/wordpress-mobile/WordPress-iOS). There are also changes to a further number of existing files, with those changes only impacting the block on the native apps.
* With this initial port, basic functionality will be available for simple WordPress.com sites. It's possible to add the block to a simple WordPress.com site, select different layouts, and preview some of them. Some functionality is still missing, such as in-editor previews and the ability to preview a square layout. Due to this missing functionality, the block is behind a DEV flag for now.
* We'll be iterating and adding further polish in future phases. Please see [Project Board](https://github.com/orgs/wordpress-mobile/projects/104/views/1) or Gutenberg Mobile issues with the "Tiled Gallery Block" tag for specifics.

#### Jetpack product discussion

* Main project thread for the Tiled Gallery block on mobile: p9ugOq-1Tb-p2
* Discussion on using inner blocks: p9dueE-3wR-p2 
* Reasoning for this PR's reduced scope: p9ugOq-2zb-p2

#### Does this pull request change what data or activity we track or use?
No, it does not.

#### Screenshots

##### Mobile

| Empty block (dark mode)  | Gallery + settings (dark mode) |
| ------------- | ------------- |
| <img src="https://user-images.githubusercontent.com/2998162/149377441-9f3bface-b9fa-4479-99e5-6dfe6ef29f84.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/149377556-19df705d-3cf1-4d17-9302-c78bc5947527.png" width="100%"> |

| Gallery with images  | Gallery preview |
| ------------- | ------------- | 
| <img src="https://user-images.githubusercontent.com/2998162/149377638-ead23b02-4005-4127-af31-a9bbab5c4be4.png" width="100%"> | <img src="https://user-images.githubusercontent.com/2998162/149362406-ad1767fe-4191-4d02-a949-7cfa36417eaf.png" width="100%"> |

##### Web

https://user-images.githubusercontent.com/1898325/148996794-77602310-2091-4eec-bc67-de27199ec081.mov



#### Testing instructions:

##### How to test these changes on mobile

We should verify that the block appears and works as expected within the apps. Note, for this iteration, the block is only available on simple WordPress.com sites and is behind a DEV flag (so it is **not** available on the installable Android and iOS builds provided on the linked PRs):

* With these branches checked out, open the editor in the WordPress Android or iOS app.
* Verify that it's possible to add the Tiled Gallery block to the editor.
* Add multiple images to the Tiled Gallery block and then select the square layout from the block's settings.
* Save and preview the post to confirm that the block displays as expected within the editor.
* If a block is added via the web, it should then be possible to edit via the app and vice versa. _(Note, there may be some quirks with layout changes when editing between the app and web at the moment, but these will be resolved in future phases.)_

##### How to test these changes on web

1. Setup a development environment
2. Check out this branch locally 
3. Test that the Tiled Gallery works as it did before on the web (check for regressions)
